### ISSUE TITLE
Enemies With Melee Based Spells Can Effect Each Other When Bunched Together

### ISSUE THREAD

**Issue by magicono43** [Interkarma/daggerfall-unity#2345]
So this basically applies to spiders and likely scorpions at least. If they attack the player (possibly other enemies as well) and are within close proximity to another entity, there is a chance one of those nearby enemies will actually be "hit" by the spell "melee spell" in question.

I have no clue how this works, as I figure stuff like "Spider Touch" is considered a touch type effect and would not have a projectile created or something. The enemies might be attacking each other technically due to the proximity, but they don't seem to hurt each other, just get effected by the paralyzing spell effect.

No clue how this might be fixed honestly as I don't really understand it currently, but figured I'd report it, as it definitely feels like a bug (no pun intended.)

Here are some images of spiders paralyzing each other trying to attack me.

![2022_05_24_19_42_51](https://user-images.githubusercontent.com/39843717/170150942-a2c791de-7a50-42e1-8aaf-c33fb8adb879.jpg)
![2022_05_24_19_53_49](https://user-images.githubusercontent.com/39843717/170150956-77ec3b94-b217-4ac7-8de0-0af1ac8e8867.jpg)

**Comment by Interkarma** [comment]
While touch isn't a projectile, it does use a raycast for striking. The spider is certainly aiming for player, but another entity in the line of attack could be hit instead. It's still no different from a spell or arrow hitting the wrong target in line of fire when aiming for player.

The combat and spell systems are honest in the sense they don't cheat and only target the player. Rather, they physically model attacks and respect the outcomes. The infighting system is built around this, as are effects that cause something to fight for player (e.g. Sanguine Rose).

I see this as a proper emergent outcome rather than a bug. There's almost no scenario where I'd want to stop this from happening, and I believe the game would be poorer if enemies could only strike player and not each other.

I'll leave open and will listen to feedback, but I'm unlikely to make a change here.

**Comment by magicono43** [comment]
Yeah, I think that's a fair assessment, I mostly thought it seemed weird and figured I may as well put a report out just in-case this behavior was unknown.

It is strange to some degree that the actual melee attack attempts themselves don't seem to act similarly, or at least the enemies don't get hurt from one another unless they are actively targeting them with melee attacks. But I'd think the spiders getting hit would then become aggressive to the other spider that hit them with the paralysis. But maybe enemies only try to in-fight if they get hit with a damage dealing spell, rather than purely effect based ones? That's just my guess at least.

**Comment by KABoissonneault** [comment]
Yeah I'm more surprised melee attacks don't get accidents. Though it's arguably more understandable, since enemies will melee in proximity a lot more

**Comment by Interkarma** [comment]
I'll look into why this seems to happen more with touch spell attacks than melee attacks. There's probably some difference in how the hits are determined making this more likely with spells. Maybe just some tuning needed to that doesn't change the desirable behaviour (i.e. anything can potentially attack anything).

**Comment by magicono43** [comment]
Thanks for looking into so far. I was honestly curious how this was actually working, and my first assumption on why was actually pretty close :P
